### ISSUE TITLE
Push notifications when app is closed

### ISSUE THREAD

**Issue by GleisonOliveira** [telegramdesktop/tdesktop#24104]
### Steps to reproduce

1. App should be closed
2. Receive a message

### Expected behaviour

App should receive push notifications when is closed.

### Actual behaviour

App does not receive any push notification when is closed.

### Operating system

Windows 11

### Version of Telegram Desktop

3.5.1

### Installation source

Microsoft Store

**Comment by hessam-kk** [comment]
are you receiving notification when app is open?

**Comment by SistemasCriativa** [comment]
Yes, when the app is open the push works well.

**Comment by GleisonOliveira** [comment]
Yes, when the app is open the push works well.

**Comment by stlr00** [comment]
Check it: [Windows Push Notification Services](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/windows-push-notification-services--wns--overview)

**Comment by GleisonOliveira** [comment]
> Check it: [Windows Push Notification Services](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/windows-push-notification-services--wns--overview)

The notifications is working well, i can receive push from another app's and from Telegram (if it's open), but i would receive notifications even if app is closed, the same way i receive from another app's like email and whatsapp.

**Comment by stlr00** [comment]
> > Check it: [Windows Push Notification Services](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/windows-push-notification-services--wns--overview)
> 
> The notifications is working well, i can receive push from another app's and from Telegram (if it's open), but i would receive notifications even if app is closed, the same way i receive from another app's like email and whatsapp.

yeah, i think telegram dont use it, and notifications work only local

**Comment by Samuelfaure** [comment]
Problem also present on Linux

**Comment by ChloeSheasby** [comment]
This problem is also present on macOS. I would love to see a good background process that could send notifications with the app being closed!

**Comment by ilya-fedin** [comment]
The application itself is a background process: it goes into background once you click the close button on titlebar. It won't work only if you're quitting the app completely but then no background process would help you if you're doing things like this literally saying "no, thanks. I don't want app to stay in background"

**Comment by GleisonOliveira** [comment]
> The application itself is a background process: it goes into background once you click the close button on titlebar. It won't work only if you're quitting the app completely but then no background process would help you if you're doing things like this literally saying "no, thanks. I don't want app to stay in background".

Looks like telegram use just local notifications, Whatsapp can receive notifications even if was closed, because use Windows Push Notification Service.
I would like to receive notifications even if I don't open Telegram, because I don't like to enable autorun apps on my system.

A native messenger without real notification is like a web app and does not make sense.

**Comment by ilya-fedin** [comment]
> A native messenger without real notification is like a web app and does not make sense.

PC IM applications were using the common (non-push) notifications for around a decade. I'm also not sure you can call tdesktop native, it's a cross-platform application written in Qt and [Qt still has no support for push notifications](https://bugreports.qt.io/browse/QTBUG-71517).

**Comment by GleisonOliveira** [comment]
> > A native messenger without real notification is like a web app and does not make sense.
> 
> PC IM applications were using the common (non-push) notifications for around a decade. I'm also not sure you can call tdesktop native, it's a cross-platform application written in Qt and [Qt still has no support for push notifications](https://bugreports.qt.io/browse/QTBUG-71517).

It seems that Telegram already has support for native Windows notifications (as screenshotted), but it doesn't seem to implement this through WNS, i.e. it's just a call to the local notifications API.

![image](https://github.com/telegramdesktop/tdesktop/assets/12660785/7e80bd92-e139-42ce-b1c7-91477380ce8e)

Using WNS, the App would only have to subscribe to a notification channel the first time it is opened, from then on the notifications are sent through the telegram servers to the servers of the Azure WNS service and it is the one who sends the notification to the system, the telegram would only have to get the information from that notification when the user clicks on it

https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/windows-push-notification-services--wns--overview

https://learn.microsoft.com/en-us/azure/notification-hubs/configure-windows-push-notification-service

The way it works is the same as what already happens with GCM on android and APNs on IOS

**Comment by ilya-fedin** [comment]
The thing is implementing this via Windows APIs won't help other platforms. And personally I'm a Linux user, I don't have interest for a Windows-only solution

**Comment by GleisonOliveira** [comment]
> The thing is implementing this via Windows APIs won't help other platforms. And personally I'm a Linux user, I don't have interest for a Windows-only solution

Windows alone accounts for more than 65% of the market, along with MacOs which holds approximately 20%, both of which support this type of notification.

![image](https://github.com/telegramdesktop/tdesktop/assets/12660785/0df4fc70-ec55-477c-9243-5760948dea30)

Not having this feature just because it is not compatible with linux is doubtful to say the least, since linux represents an insignificant part when it comes to desktops.

It's not a windows-only feature, but rather a feature where linux would be the only one not to have.

**Comment by ilya-fedin** [comment]
@GleisonOliveira I'm an OSS contributor and I work on what I have interest in.
